File: cqim-app/server/public-url.ts
const DEFAULT_PUBLIC_BASE_URL = 'https://cq.je';

/** Canonical public origin for invite links, OG tags, and push assets. */
export function getPublicBaseUrl(): string {
  const raw = process.env.PUBLIC_BASE_URL?.trim();
  if (raw) return raw.replace(/\/$/, '');
  return DEFAULT_PUBLIC_BASE_URL;
}

export function publicUrl(path = ''): string {
  const base = getPublicBaseUrl();
  if (!path) return base;
  if (/^https?:\/\//i.test(path)) return path;
  return `${base}${path.startsWith('/') ? path : `/${path}`}`;
}

File: cqim-app/go-server/internal/handler/media/publicurl.go
package media

// publicurl.go 移植自 server/public-url.ts。

import (
	"os"
	"regexp"
	"strings"
)

// defaultPublicBaseURL 默认公网域名（对应 TS DEFAULT_PUBLIC_BASE_URL）。
const defaultPublicBaseURL = "https://cq.je"

var absoluteURLRe = regexp.MustCompile(`(?i)^https?://`)

// GetPublicBaseURL 规范公网 origin（对应 TS getPublicBaseUrl）。
func GetPublicBaseURL() string {
	raw := strings.TrimSpace(os.Getenv("PUBLIC_BASE_URL"))
	if raw != "" {
		return strings.TrimSuffix(raw, "/")
	}
	return defaultPublicBaseURL
}

// PublicURL 拼接公网完整 URL（对应 TS publicUrl）。
func PublicURL(path string) string {
	base := GetPublicBaseURL()
	if path == "" {
		return base
	}
	if absoluteURLRe.MatchString(path) {
		return path
	}
	if strings.HasPrefix(path, "/") {
		return base + path
	}
	return base + "/" + path
}
